### ISSUE TITLE
🐛 [SAMBA NAS] Getting error 100093 when trying to add a file via SMB on an external exFat disk attached to the pi

### ISSUE THREAD

**Issue by 4rianton** [dianlight/hassio-addons#175]
### Addon

SambaNAS

### Description

Trying to add files to the mounted drive via SMB results in error 100093. Writing to Raspberry Pi's card via SMB works without issues. 

<img width="516" alt="Screenshot 2023-10-27 at 15 57 08" src="https://github.com/dianlight/hassio-addons/assets/7961281/29172b39-2f7c-44af-b81e-944b75377e5b">

### Reproduction steps

```bash
1. Have Homeassistant OS and SAMBA NAS installed
2. have an external disk connected (exFat formatted in my case)
3. have it configured in the moredisks section
4. connect to SMB (using mac in my case)
5. try to add a small folder - it succeeds
6. try to add a bigger file - Fails with 100093
7. mount a different folder from Pi's internal storage on SMB (like "addons") - can add files without issues
```

### Addon Logs

```bash
s6-rc: info: service s6rc-oneshot-runner: starting
s6-rc: info: service s6rc-oneshot-runner successfully started
s6-rc: info: service base-addon-banner: starting

-----------------------------------------------------------
 Add-on: Samba NAS
 Expose Home Assistant disc with SMB/CIFS
-----------------------------------------------------------
 Add-on version: 10.0.2-nas4
 You are running the latest version of this add-on.
 System: Home Assistant OS 11.1  (aarch64 / raspberrypi4-64)
 Home Assistant Core: 2023.10.5
 Home Assistant Supervisor: 2023.10.1
-----------------------------------------------------------
 Please, share the above information when looking for help
 or support in, e.g., GitHub, forums or the Discord chat.
-----------------------------------------------------------
s6-rc: info: service base-addon-banner successfully started
s6-rc: info: service fix-attrs: starting
s6-rc: info: service base-addon-log-level: starting
s6-rc: info: service fix-attrs successfully started
s6-rc: info: service base-addon-log-level successfully started
s6-rc: info: service legacy-cont-init: starting
s6-rc: info: service legacy-cont-init successfully started
s6-rc: info: service init-automount: starting
[15:39:33] INFO: Protection Mode is false
[15:39:33] INFO: MoreDisk or Automount option found!
---------------------------------------------------
Supported fs: 	ext3	ext2	ext4	squashfs	vfat	msdos	f2fs	fuseblk	ntfs3	exfat	btrfs	xfs
Supported fusefs: 
---------------------------------------------------
[15:39:33] INFO: MediaLibrary disabled in config. Disk are mounted only for this addon!
---------------------------------------------------
[15:39:34] INFO: Available Disk Labels:
[15:39:34] INFO: 	EFI[vfat]
[15:39:34] INFO: 	MEDIA[exfat]
---------------------------------------------------
[15:39:34] INFO: MoreDisks to mount:

	MEDIA
[15:39:34] INFO: Automount is Enabled!
---------------------------------------------------
[15:39:34] INFO: Mounting disks:

	EFI
	MEDIA
---------------------------------------------------
[15:39:34] WARNING: Your EFI is vfat. Permissions and ACL don't works and this is an EXPERIMENTAL support
[15:39:34] INFO: Mount EFI[vfat] Success!
[15:39:34] WARNING: Your MEDIA is exfat. Permissions and ACL don't works and this is an EXPERIMENTAL support
[15:39:34] INFO: Mount MEDIA[exfat] Success!
s6-rc: info: service init-automount successfully started
s6-rc: info: service init-smartd: starting
s6-rc: info: service init-samba: starting
s6-rc: info: service hd-idle: starting
s6-rc: info: service hd-idle successfully started
[15:39:35] INFO: Enabling S.M.A.R.T for /dev/sda
smartctl 7.3 2022-02-28 r5338 [aarch64-linux-6.1.21-v8] (local build)
Copyright (C) 2002-22, Bruce Allen, Christian Franke, www.smartmontools.org

[15:39:35] INFO: Hostname: homeassistant
=== START OF ENABLE/DISABLE COMMANDS SECTION ===
SMART Enabled.
SMART Attribute Autosave Enabled.
SMART Automatic Timers not supported

SMART Automatic Offline Testing Enabled every four hours.

s6-rc: info: service init-smartd successfully started
[15:39:36] INFO: Interfaces: wlan0 end0 
[15:39:36] INFO: Docker Interface: hassio 172.30.32.0/23
tdbsam_open: Converting version 0.0 database to version 4.0.
tdbsam_convert_backup: updated /var/lib/samba/private/passdb.tdb file.
Added user home.
---------------------------------------------------
[15:39:37] INFO: Exposed Disks Summary:
[config]                path = /config # TM:false#
[addons]                path = /addons # TM:false#
[ssl]                   path = /ssl # TM:false#
[share]                 path = /share # TM:false#
[backup]                path = /backup # TM:false#
[media]                 path = /media # TM:false#
[EFI]                   path = /mnt/EFI # TM:true#
[MEDIA]                 path = /mnt/MEDIA # TM:true#
---------------------------------------------------
s6-rc: info: service init-samba successfully started
s6-rc: info: service init-mqtt: starting
s6-rc: info: service wsdd: starting
s6-rc: info: service smbd: starting
s6-rc: info: service nmbd: starting
s6-rc: info: service avahi: starting
s6-rc: info: service wsdd successfully started
s6-rc: info: service smbd successfully started
s6-rc: info: service nmbd successfully started
s6-rc: info: service cifs-supervisor-mount: starting
s6-rc: info: service avahi successfully started
[15:39:37] INFO: Starting the AVAHI for homeassistant...
Waiting for daemon ...
s6-rc: info: service cifs-supervisor-mount successfully started
[15:39:37] INFO: MQTT support not enabled in config
s6-rc: info: service init-mqtt successfully started
s6-rc: info: service mqtt-disk-handler: starting
[15:39:37] INFO: Starting the wsdd daemon for WORKGROUP/homeassistant...
s6-rc: info: service mqtt-disk-handler successfully started
s6-rc: info: service mqtt-handler: starting
s6-rc: info: service mqtt-handler successfully started
s6-rc: info: service legacy-services: starting
s6-rc: info: service legacy-services successfully started
[15:39:37] INFO: Interfaces: -i wlan0 -i end0 
2023-10-27 15:39:38,412:wsdd INFO(pid 344): using pre-defined UUID cb6a33a0-b841-531d-a34e-09ecec492eb1
2023-10-27 15:39:38,415:wsdd INFO(pid 344): joined multicast group 239.255.255.250 on 192.168.1.14%end0
2023-10-27 15:39:38,421:wsdd INFO(pid 344): scheduling Hello message via end0 to 239.255.255.250%end0
2023-10-27 15:39:38,545:wsdd INFO(pid 344): joined multicast group [ff02::c] on fe80::78bd:f388:b28f:6cde%end0
2023-10-27 15:39:38,547:wsdd INFO(pid 344): scheduling Hello message via end0 to ff02::c%end0
nmbd version 4.18.4 started.
Copyright Andrew Tridgell and the Samba Team 1992-2023
smbd version 4.18.4 started.
Copyright Andrew Tridgell and the Samba Team 1992-2023
INFO: Profiling support unavailable in this build.
ad_convert_xattr: SMB_VFS_CREATE_FILE failed
fruit_freaddir_attr: ad_convert("Playables") failed
*****

Samba name server HOMEASSISTANT is now a local master browser for workgroup WORKGROUP on subnet 192.168.1.14

*****
*****

Samba name server HOMEASSISTANT is now a local master browser for workgroup WORKGROUP on subnet 172.30.32.1

*****



.................


ad_convert_xattr: SMB_VFS_CREATE_FILE failed
fruit_freaddir_attr: ad_convert("Playables/The video/video.avi") failed
ad_convert_xattr: SMB_VFS_CREATE_FILE failed
fruit_freaddir_attr: ad_convert("Playables/The video/video.avi") failed
```

### Addon Config

```bash
workgroup: WORKGROUP
username: user
password: pwd
allow_hosts:
  - 10.0.0.0/8
  - 172.16.0.0/12
  - 192.168.0.0/16
  - 169.254.0.0/16
  - fe80::/10
  - fc00::/7
  - 192.168.1.1/255
automount: true
moredisks:
  - MEDIA1
available_disks_log: true
medialibrary:
  enable: false
veto_files:
  - ._*
  - .DS_Store
  - Thumbs.db
  - icon?
  - .Trashes
compatibility_mode: false
wsdd2: false
mqtt_use_legacy_entities: false
autodiscovery: {}
other_users: []
acl: []
interfaces: []
```

### Architecture

aarch64

### OS

HaOS

**Comment by dianlight** [comment]
should be fixed  in the release 12.1.0-nas

**Comment by 4rianton** [comment]
Still fails for me: 

```
`---------------------------------------------------
[20:07:19] INFO: Exposed Disks Summary:
[config]                path = /homeassistant # TM:false#
[addons]                path = /addons # TM:false#
[ssl]                   path = /ssl # TM:false#
[share]                 path = /share # TM:false#
[backup]                path = /backup # TM:false#
[media]                 path = /media # TM:false#
[addon_configs]         path = /addon_configs # TM:false#
[EFI]                   path = /mnt/EFI # TM:true#
[MEDIA]                 path = /mnt/MEDIA # TM:true#
---------------------------------------------------
s6-rc: info: service init-samba successfully started
s6-rc: info: service init-mqtt: starting
s6-rc: info: service wsdd: starting
s6-rc: info: service smbd: starting
s6-rc: info: service nmbd: starting
s6-rc: info: service avahi: starting
s6-rc: info: service wsdd successfully started
s6-rc: info: service smbd successfully started
s6-rc: info: service nmbd successfully started
s6-rc: info: service cifs-supervisor-mount: starting
s6-rc: info: service avahi successfully started
s6-rc: info: service cifs-supervisor-mount successfully started
[20:07:19] INFO: Starting the AVAHI for homeassistant...
Waiting for daemon ...
[20:07:20] INFO: MQTT support not enabled in config
s6-rc: info: service init-mqtt successfully started
s6-rc: info: service mqtt-disk-handler: starting
s6-rc: info: service mqtt-disk-handler successfully started
s6-rc: info: service mqtt-handler: starting
s6-rc: info: service mqtt-handler successfully started
s6-rc: info: service legacy-services: starting
[20:07:20] INFO: Starting the wsdd daemon for WORKGROUP/homeassistant...
s6-rc: info: service legacy-services successfully started
[20:07:20] INFO: Interfaces: -i end0 -i wlan0 
2023-11-17 20:07:20,850:wsdd INFO(pid 350): using pre-defined UUID cb6a33a0-b841-531d-a34e-09ecec492eb1
2023-11-17 20:07:20,853:wsdd INFO(pid 350): joined multicast group 239.255.255.250 on 192.168.1.14%end0
2023-11-17 20:07:20,859:wsdd INFO(pid 350): scheduling Hello message via end0 to 239.255.255.250%end0
2023-11-17 20:07:22,967:wsdd INFO(pid 350): joined multicast group [ff02::c] on fe80::78bd:f388:b28f:6cde%end0
2023-11-17 20:07:22,969:wsdd INFO(pid 350): scheduling Hello message via end0 to ff02::c%end0
smbd version 4.18.8 started.
nmbd version 4.18.8 started.
Copyright Andrew Tridgell and the Samba Team 1992-2023
Copyright Andrew Tridgell and the Samba Team 1992-2023
INFO: Profiling support unavailable in this build.
ad_convert_xattr: SMB_VFS_CREATE_FILE failed
fruit_freaddir_attr: ad_convert("Playables") failed
ad_convert_xattr: SMB_VFS_CREATE_FILE failed
fruit_freaddir_attr: ad_convert("Playables") failed
ad_convert_xattr: SMB_VFS_CREATE_FILE failed
fruit_freaddir_attr: ad_convert("Playables") failed
ad_convert_xattr: SMB_VFS_CREATE_FILE failed
fruit_freaddir_attr: ad_convert("Playables") failed
*****

Samba name server HOMEASSISTANT is now a local master browser for workgroup WORKGROUP on subnet 192.168.1.14

*****
*****

Samba name server HOMEASSISTANT is now a local master browser for workgroup WORKGROUP on subnet 172.30.32.1

*****
ad_convert_xattr: SMB_VFS_CREATE_FILE failed
fruit_freaddir_attr: ad_convert("Playables/Succession/S04/Succession_S04E08_America_Decides_1080p_WEB_HD_x265_10Bit_6CH_Pahe.mkv") failed
ad_convert_xattr: SMB_VFS_CREATE_FILE failed

````

**Comment by dianlight** [comment]
Try to disable timemachine. vfat and exfat don't play well with extended attributes.

```yaml
acl:
  - share: EFI
     timemachine: false
  - share: MEDIA
     timemachine: false
```

L.

**Comment by dianlight** [comment]
Fixed in 12.2.0-nas #191
